comp110-worksheet-6/OxoBoard.cs

```diff
@@ -7,42 +7,210 @@
+        public OxoBoard()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            boardArray = new Mark[3, 3];
+            // 2 For loops used to go through each row and collum of board
+            for (int x = 0; x < 3; x++)
+            {
+                for (int y = 0; y < 3; y++)
+                {
+                    // Then for each tile of the board it is set as empty
+                    boardArray[x, y] = Mark.None;
+                }
+            }
 		}
 
 		// Return the contents of the specified square.
 		public Mark GetSquare(int x, int y)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            if(boardArray[x, y] != null)
+            {
+                // Returns what it inside the tile being checked
+                return boardArray[x, y];
+            }
+            else
+            {
+                // Returns that the tile is empty if cords are outside the board
+                return Mark.None;
+            }
+           
+           
+
+        
+
 		}
 
 		// If the specified square is currently empty, fill it with mark and return true.
 		// If the square is not empty, leave it as-is and return False.
 		public bool SetSquare(int x, int y, Mark mark)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            // If statement that checks if cords are inside the board
+            if((x >= 0 && x <= 2) && (y >= 0 && y <= 2))
+            {
+                if (boardArray[x, y] == Mark.None)
+                {
+                    boardArray[x, y] = mark;
+                    return true;
+                    // Returns true if board tile was empty and was now replaced by X or O
+                }
+                else
+                {
+                    return false;
+                    // Returns flase if tile was already occupied and leaves tile as is
+                }
+            }
+            else
+            {
+                return false;
+                // Returns false as invalid move if player inputted cords that were off the 3x3 grid
+            }
+          
+            
+
+            
+
 		}
 
 		// If there are still empty squares on the board, return false.
 		// If there are no empty squares, return true.
 		public bool IsBoardFull()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+            int filledTiles = 0;
+            // Int set for amount of tiles that are filled
+
+            // For loops used again to check every tile on the 3x3 grid
+            for (int x = 0; x < 3; x++)
+            {
+                for (int y = 0; y < 3; y++)
+                {
+                    if(boardArray[x , y] == Mark.None)
+                    {
+                        // Board is empty on this tile
```

Comment:
You could `return false` at this point -- as soon as you've found an empty tile, you know the board is not full